### PR TITLE
Provide warning for unsupported rest-client features in native

### PR DESCRIPTION
**Pull request by geoand**
This doesn't fix anything, but does at the least warn users building
a native binary that rest-client with `@ClientHeaderParam`
and default methods may not work.

Relates to #10249

### CODE REVIEW

**Review by kenfinnigan — APPROVED** [review]

**Comment by gsmet** [comment]
I killed the CI builds to get resources for PRs critical for CR1. We will have to restart them later.

**Comment by geoand** [comment]
Sure thing

**Comment by gsmet** [comment]
I restarted CI.

File: extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java [hunk 1]
```diff
@@ -220,6 +227,40 @@ public void register(RegistrationContext registrationContext) {
         extensionSslNativeSupport.produce(new ExtensionSslNativeSupportBuildItem(Feature.REST_CLIENT));
     }
 
+    // currently default methods on a rest-client interface
+    // that is annotated with ClientHeaderParam
+    // leads to NPEs (see https://github.com/quarkusio/quarkus/issues/10249)
+    // so lets warn users about it's use
```
Comment (gsmet) [inline extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java:233]:
```suggestion
    // so lets warn users about its use
```

File: extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java [hunk 2]
```diff
@@ -220,6 +227,40 @@ public void register(RegistrationContext registrationContext) {
...
+    private void warnAboutNotWorkingFeaturesInNative(PackageConfig packageConfig, Map<DotName, ClassInfo> interfaces) {
+        if (!packageConfig.type.equalsIgnoreCase(PackageConfig.NATIVE)) {
+            return;
+        }
+        Set<DotName> dotNames = new HashSet<>();
+        for (ClassInfo intefaze : interfaces.values()) {
```
Comment (gsmet) [inline extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java:239]:
```suggestion
        for (ClassInfo interfaze : interfaces.values()) {
```

**Comment by geoand** [comment]
Thanks!

File: extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java [hunk 3]
```diff
@@ -220,6 +227,40 @@ public void register(RegistrationContext registrationContext) {
...
+                    }
+                }
+                if (hasDefault) {
+                    dotNames.add(intefaze.name());
+                }
+            }
+        }
+        if (!dotNames.isEmpty()) {
+            log.warnf("rest-client interfaces that contain default methods and are annotated with '@" + CLIENT_HEADER_PARAM
+                    + "' might not work properly in native-mode. Offending interfaces are: "
```
Comment (gsmet) [inline extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java:255]:
```suggestion
                    + "' might not work properly in native mode. Offending interfaces are: "
```

File: extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java [hunk 4]
```diff
@@ -220,6 +227,40 @@ public void register(RegistrationContext registrationContext) {
...
+                    + dotNames.stream().map(d -> "'" + d.toString() + "'").collect(Collectors.joining(",")));
```
Comment (gsmet) [inline extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java:256]:
```suggestion
                    + dotNames.stream().map(d -> "'" + d.toString() + "'").collect(Collectors.joining(", ")));
```

**Review by gsmet — DISMISSED** [review]
I added some very small suggestions, feel free to ignore.

**Comment by geoand** [comment]
Thanks for the suggestions, I applied them

**Pull request merged**